### PR TITLE
Enhancements to plugins list command

### LINKED ISSUE #489: zowe plugins list - alphabetical order
Currently, it appears that `zowe plugins list` returns the list of installed plugins in the order in which they were installed. Over time this list may become large and difficult to read through. I suggest we order the list alphabetically by pluginName.

### PR DESCRIPTION
**Pull request by t1m0thyj**
Resolves #489 and #743

### CODE REVIEW

File: packages/imperative/src/plugins/cmd/list/list.handler.ts [hunk 1]
```diff
@@ -48,21 +48,25 @@ export default class ListHandler implements ICommandHandler {
...
+                    listOutput = listOutput + `${chalk.yellow.bold(" -- pluginName: ")}` +
+                        `${chalk.red.bold(pluginName)} \n`;
+                    listOutput = listOutput + `${chalk.yellow.bold(" -- package: ")}` +
+                        `${chalk.red.bold(installedPlugins[pluginName].package)} \n`;
+                    listOutput = listOutput + `${chalk.yellow.bold(" -- version: ")}` +
+                        `${chalk.red.bold(installedPlugins[pluginName].version)} \n`;
+                    listOutput = listOutput + `${chalk.yellow.bold(" -- registry: ")}` +
+                        installedPlugins[pluginName].registry + "\n\n";
+                } else {
+                    listOutput += `${chalk.yellow(pluginName)}@${installedPlugins[pluginName].version}\n`;
```
Comment (t1m0thyj) [inline packages/imperative/src/plugins/cmd/list/list.handler.ts:68]: Not sure whether to show `pluginName@version` or `package@version`? Can't remember seeing a case where `pluginName` and `package` are different.

Comment (gejohnston) [reply packages/imperative/src/plugins/cmd/list/list.handler.ts:68]: Perhaps they are always the same. If they are ever different, a user who does a 'plugin list' is to get the correct name to be able to run 'plugin uninstall' would probably need the package name. Can you think of other use cases in which the pluginName would be the needed output?

**Review by gejohnston — APPROVED** [review]

File: packages/imperative/src/plugins/cmd/list/list.definition.ts [hunk 2]
```diff
@@ -21,5 +21,13 @@ export const listDefinition: ICommandDefinition = {
     type: "command",
     summary: "List installed plug-ins",
     description: "List all plug-ins installed.",
-    handler: join(__dirname, "list.handler")
+    handler: join(__dirname, "list.handler"),
+    options: [
+        {
+            name: "short",
+            aliases: ["-s"],
```
Comment (awharn) [inline packages/imperative/src/plugins/cmd/list/list.definition.ts:28]: Is there a reason this is -s instead of s?

**Review by awharn — COMMENTED** [review]
One question

File: packages/imperative/src/plugins/cmd/list/list.definition.ts (hunk 2 above)
Comment (t1m0thyj) [reply packages/imperative/src/plugins/cmd/list/list.definition.ts]: Good catch, fixed

File: packages/imperative/src/plugins/cmd/list/list.handler.ts (hunk 1 above)
Comment (t1m0thyj) [reply packages/imperative/src/plugins/cmd/list/list.handler.ts:68]: It looks like the `plugins uninstall` and `plugins update` commands assume their positional argument is present as a key in the plugins JSON file. So using `pluginName` in this case which is also a key in the plugins JSON file makes sense.

**Review by awharn — APPROVED** [review]

**Pull request merged**